src/runtime/device_interface.cpp

```diff
@@ -180,21 +186,30 @@ WEAK int halide_device_sync(void *user_context, struct buffer_t *buf) {
         interface = halide_get_device_interface(buf->dev);
     }
     if (interface == NULL) {
-        return -1;
+        return halide_error_code_no_device_interface;
+    }
+    int result = interface->device_sync(user_context, buf);
+    if (result) {
+        return halide_error_code_device_sync_failed;
+    } else {
+        return 0;
     }
-    return interface->device_sync(user_context, buf);
 }
 
 /** Allocate device memory to back a buffer_t. */
 WEAK int halide_device_malloc(void *user_context, struct buffer_t *buf, const halide_device_interface *interface) {
     const halide_device_interface *current_interface = halide_get_device_interface(buf->dev);
-    debug(user_context) << "halide_device_malloc: " << buf << " interface " << interface <<
-        " host: " << buf->host << ", dev: " << buf->dev << ", host_dirty: " << buf->host_dirty << ", dev_dirty:" << buf->dev_dirty <<
-        " buf current interface: " << current_interface << "\n";
+    debug(user_context) << "halide_device_malloc: " << buf
+                        << " interface " << interface
+                        << " host: " << buf->host
+                        << ", dev: " << buf->dev
+                        << ", host_dirty: " << buf->host_dirty
+                        << ", dev_dirty:" << buf->dev_dirty
+                        << " buf current interface: " << current_interface << "\n";
 
     // halide_device_malloc does not support switching interfaces.
     if (current_interface != NULL && current_interface != interface) {
-        return -1;
+        return halide_error_code_device_malloc_failed;
```

Comment:
There's a `halide_error_code_no_device_interface` used in a few situations above, should there also be a `halide_error_code_bad_device_interface` or something?